portable/GCC/MSP430FR5969/port.c

```diff
@@ -0,0 +1,320 @@
+not the MCLK. */
+#define portACLK_FREQUENCY_HZ           ( ( TickType_t ) 32768 )
+#define portINITIAL_CRITICAL_NESTING    ( ( uint16_t ) 10 )
+#define portFLAGS_INT_ENABLED           ( ( StackType_t ) 0x08 )
+
+/* We require the address of the pxCurrentTCB variable, but don't want to know
+any details of its type. */
+typedef void TCB_t;
+extern volatile TCB_t * volatile pxCurrentTCB;
+
+/* Most ports implement critical sections by placing the interrupt flags on
+the stack before disabling interrupts.  Exiting the critical section is then
+simply a case of popping the flags from the stack.  As mspgcc does not use
+a frame pointer this cannot be done as modifying the stack will clobber all
+the stack variables.  Instead each task maintains a count of the critical
+section nesting depth.  Each time a critical section is entered the count is
+incremented.  Each time a critical section is left the count is decremented -
+with interrupts only being re-enabled if the count is zero.
+
+usCriticalNesting will get set to zero when the scheduler starts, but must
+not be initialised to zero as this will cause problems during the startup
+sequence. */
+volatile uint16_t usCriticalNesting = portINITIAL_CRITICAL_NESTING;
+/*-----------------------------------------------------------*/
+
+/*
+ * Macro to save a task context to the task stack.  This simply pushes all the
+ * general purpose msp430 registers onto the stack, followed by the
+ * usCriticalNesting value used by the task.  Finally the resultant stack
+ * pointer value is saved into the task control block so it can be retrieved
+ * the next time the task executes.
+ */
+#ifdef __LARGE_DATA_MODEL__
+#define portSAVE_CONTEXT()                                   \
+    asm volatile (  "pushm.a #12, r15                 \n\t"  \
+                    "movx.w &usCriticalNesting, r14   \n\t"  \
+                    "pushm.a #1, r14                  \n\t"  \
+                    "movx.a &pxCurrentTCB, r12        \n\t"  \
+                    "movx.a sp, 0(r12)                \n\t"  \
+                );
+#else
+#define portSAVE_CONTEXT()                                   \
+    asm volatile (  "pushm.w #12, r15                 \n\t"  \
+                    "mov.w &usCriticalNesting, r14    \n\t"  \
+                    "push.w r14                       \n\t"  \
+                    "mov.w &pxCurrentTCB, r12         \n\t"  \
+                    "mov.w sp, 0(r12)                 \r\t"  \
+                );
+#endif
+
+/*
+ * Macro to restore a task context from the task stack.  This is effectively
+ * the reverse of portSAVE_CONTEXT().  First the stack pointer value is
+ * loaded from the task control block.  Next the value for usCriticalNesting
+ * used by the task is retrieved from the stack - followed by the value of all
+ * the general purpose msp430 registers.
+ *
+ * The bic instruction ensures there are no low power bits set in the status

```

Comment:
Looks like the bic instruction has been removed in this port.

Reply:
I removed this comment. These bits are cleared in `prvTickISR` using `__bic_SR_register_on_exit`.